atlas/packages/app/src/shared/components/Button/Button.tsx

```diff
@@ -1,31 +1,44 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { SerializedStyles } from '@emotion/core'
-import { ButtonStyleProps, useCSS } from './Button.style'
-import { BlockIcon } from '../../icons'
+import { ButtonStyleProps, StyledButton, StyledIcon } from './Button.style'
 
-type ButtonProps = {
-  children?: React.ReactNode
-  icon?: boolean
-  disabled?: boolean
+export type ButtonProps = {
+  icon: boolean
+  disabled: boolean
+  toggle: boolean
   containerCss: SerializedStyles
-  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
-} & ButtonStyleProps
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
+} & Omit<ButtonStyleProps, 'clickable' | 'hasText'>
 
 const Button: React.FC<Partial<ButtonProps>> = ({
```

Comment:
Since we're refactoring button now, I believe we should get rid of `Partial` here as well

Reply:
I would argue that `Partial` is beneficial here, all props are optional when the button is used elsewhere, but `ButtonStyleProps` for example should be strictly typed, so one is forced to define defaults for all style properties

Reply:
We can just as easily make the props optional the explicit way - with `?`. This way the code is more easily readable and we don't need to refactor in the future if we want to add just one required prop